services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/models/AtomicCounterRecord.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.functionaltests.models;
+
+import software.amazon.awssdk.enhanced.dynamodb.extensions.annotations.DynamoDbAtomicCounter;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+
+@DynamoDbBean
+public class AtomicCounterRecord {
+    private String id;
+    private Long defaultCounter;
+    private Long customCounter;
+    private Long decreasingCounter;
+    private String attribute1;
+
+    @DynamoDbPartitionKey
+    public String getId() {
+        return id;
+    }
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    @DynamoDbAtomicCounter
```

Comment:
Can we also add tests for other types of table schemas?

Reply:
The `AtomicCounterExtensionTest` uses a `StaticTableSchema`

Reply:
Yeah, I meant other table schemas like the ImmutableTableSchema and StaticImmutableTableSchema

Reply:
A really good point. I looked in the test code and it seems like we are in the habit of testing with StaticTableSchema and BeanTableSchema in general, and skipping ImmutableTableSchema. 

Both StaticTableSchema and BeanTableSchema wrap StaticImmutableTableSchema, therefore IMO testing those also test StaticImmutableTableSchema. 

It's really great that we have a lot of tests for the enhanced client, but I do find that the sheer number of tests make it confusing to find relevant tests as well as making sure we are covering all features as well as permutations of them. In this case (AtomicCounters), what the schemas do is that they should interpret tags correctly, so I created a new test that test custom metadata tags for all 3 schemas users tend to use directly. We have three different types of custom metadata right now so I tested each of them.

Reply:
Nice, thanks for creating parameterized tests. Yeah, I had not discovered the tests either before Ben told me about them. I wonder if there's a way to enforce the tests so that we don't forget in the future. Currently, it's purely relying on code reviews and tribal knowledge